File: Sorts & Search & Shuffling Algorithms/Fisher–Yates shuffle/Fisher_Yates.cpp
#include <vector>
#include <cstdlib>
#include <ctime>
using namespace std;

class Fisher_Yates{
public:
    template <typename T>
    static void Shuffle(vector<T> &it){
        srand(time(NULL));
        for (int i = it.size() - 1; i > 0; i--){
            int j = rand() % (i + 1);
            T temp=it[i];
            it[i]=it[j];
            it[j]=temp;
        }
    }
};

File: Sorts & Search & Shuffling Algorithms/Fisher–Yates shuffle/Fisher_Yates.java
import java.util.ArrayList;
import java.util.Random;

public class Fisher_Yates {
    public static <T extends Comparable<T>> void Shuffle(ArrayList<T> iterable) {
        Random R=new Random();
        for (int i=iterable.size()-1; i>0; i--){
            int random = R.nextInt(i+1);

            T temp= iterable.get(i);
            iterable.set(i, iterable.get(random));
            iterable.set(random,temp);
        }
    }
}
